ocs_ci/ocs/resources/mcg.py

```diff
@@ -798,21 +798,30 @@ def _check_mirroring():
                         results.append(True)
                     else:
                         results.append(False)
-
-            return all(results)
-
-        try:
-            for mirroring_is_complete in TimeoutSampler(timeout, 5, _check_mirroring):
-                if mirroring_is_complete:
-                    logger.info("All objects mirrored successfully.")
-                    return True
-                else:
-                    logger.info("Waiting for the mirroring process to finish...")
-        except TimeoutExpiredError:
-            logger.error(
-                "The mirroring process did not complete within the time limit."
-            )
-            assert False
+            current_percentage = (results.count(True) / len(results)) * 100
+            return current_percentage
+
+        mirror_percentage = _get_mirroring_percentage()
+        logger.info(f"{mirror_percentage}% mirroring is done.")
+        previous_percentage = 0
+        while mirror_percentage < 100:
+            previous_percentage = mirror_percentage
+            try:
+                for mirror_percentage in TimeoutSampler(
+                    timeout, 5, _get_mirroring_percentage
+                ):
+                    if previous_percentage == mirror_percentage:
+                        logger.error("The mirroring process is stuck.")
+                    else:
+                        break
+            except TimeoutExpiredError:
+                logger.error(
+                    f"The mirroring process is stuck from last {timeout} seconds."
+                )
+                assert False
+            mirror_percentage = _get_mirroring_percentage()
+        logger.info("All objects mirrored successfully.")
+        return True
```

Comment:
Looking at the current usage of the method, we can remove this `return True` line. Please also update the docstring that this method now throws an exception in case mirroring was not done in time instead of returning a boolean value.